__tests__/app.test.js

```diff
@@ -149,6 +149,78 @@ describe("GET 200: /api/articles", () =>{
+    
+    return request(app).post("/api/articles/4/comments").send(comment).expect(201).then((response) =>{
+      const comments = response.body
+      expect(comments[0]).toHaveProperty("comment_id", expect.any(Number));
+        expect(comments[0]).toHaveProperty("votes", expect.any(Number));
+        expect(comments[0]).toHaveProperty("author", expect.any(String));
+        expect(comments[0]).toHaveProperty("body", expect.any(String))
+        expect(comments[0]).toHaveProperty("created_at", expect.any(String));
+        expect(comments[0]).toHaveProperty("article_id", expect.any(Number))
+      expect(comments[0].body).toBe("this is a comment")
+      expect(comments[0].author).toBe("butter_bridge")
+
+      return request(app).get("/api/articles/4/comments")
+    }).then((response)=>{
+      const comments = response.body
+      expect(comments[0]).toHaveProperty("comment_id", expect.any(Number));
+        expect(comments[0]).toHaveProperty("votes", expect.any(Number));
+        expect(comments[0]).toHaveProperty("author", expect.any(String));
+        expect(comments[0]).toHaveProperty("body", expect.any(String))
+        expect(comments[0]).toHaveProperty("created_at", expect.any(String));
+        expect(comments[0]).toHaveProperty("article_id", expect.any(Number))
+      expect(comments[0].body).toBe("this is a comment")
+      expect(comments[0].author).toBe("butter_bridge")
+    })
+  })
+  test("POST 400: returns status code 400 and bad request message if sent an invalid id input", ()=>{
+    const comment = {username: "butter_bridge", body:"this is a comment"}
+    
+    return request(app).post("/api/articles/hello/comments").expect(400).send(comment).then(({ body }) => {
+      const { msg } = body;
+      expect(msg).toBe("bad request");
+    });
+  })
+  test("POST 404: returns a 404 code when a correct id is inputted but theres no item matching in the database", () =>{
+    
+    const comment = {username: "butter_bridge", body:"this is a comment"}
+    return request(app)
+    .post("/api/articles/999999/comments")
+    .expect(404)
+    .send(comment)
+    .then(({ body }) => {
+      const { msg } = body;
+     
+      expect(msg).toBe("not found");
+    });
+  })
+
+  test("POST 400:returns status code 400 and a bad request message if no comment is sent", () =>{
+    return request(app).post("/api/articles/4/comments").expect(400).then(({ body }) => {
+      const { msg } = body;
+      expect(msg).toBe("bad request");
+    });
+
+  })
+  test("POST 404: returns status code 404 and a not found message when a comment is inserting with an unknown user name", () =>{
+    const comment = {username: "person1", body:"this is a comment"}
+    return request(app)
+    .post("/api/articles/999999/comments")
+    .expect(404)
+    .send(comment)
+    .then(({ body }) => {
+      const { msg } = body;
+     
+      expect(msg).toBe("not found");
+    });
+  })
+  
+})
+
```

Comment:
Some great other tests here :) please also include another 201 test for when somebody posts the correct keys in their post but includes extra keys. Theses extra keys should be ignored and the comment posted